applications/concept-catalogue/src/main/kotlin/no/begrepskatalog/conceptregistration/BegreperApiImplK.kt

```diff
@@ -73,98 +79,29 @@ class BegreperApiImplK(val sqlStore: SqlStore) : BegreperApi {
-        }
-        if (source.kildebeskrivelse != null) {
-            if (destination.kildebeskrivelse == null) {
-                destination.kildebeskrivelse = Kildebeskrivelse()
-            }
-            if (source.kildebeskrivelse.forholdTilKilde!= null) {
-                destination.kildebeskrivelse.forholdTilKilde= source.kildebeskrivelse.forholdTilKilde
-            }
-            if (source.kildebeskrivelse.kilde!= null ) {
-                destination.kildebeskrivelse.kilde = source.kildebeskrivelse.kilde
+        if (patchedBegrep.status == Status.UTKAST) {
+            sqlStore.saveBegrep(patchedBegrep)
+            return ResponseEntity.ok(patchedBegrep)
+        } else {
+            if (isValidBegrep(patchedBegrep)) {
+                sqlStore.saveBegrep(patchedBegrep)
+                logger.info("Begrep $patchedBegrep.id has passed validation for non draft begrep and has been saved ")
+                return ResponseEntity.ok(patchedBegrep)
             }
         }
-        if (source.merknad != null) {
-            destination.merknad = source.merknad
-        }
-        if (source.eksempel != null) {
-            destination.eksempel = source.eksempel
-        }
-        if (source.fagområde != null) {
-            destination.fagområde = source.fagområde
-        }
-        if (source.bruksområde != null && source.bruksområde.isNotEmpty()) {
-            destination.bruksområde = source.bruksområde
-        }
-        if (source.kontaktpunkt != null) {
-            if (source.kontaktpunkt?.harEpost != null)
-                destination.kontaktpunkt.harEpost = source.kontaktpunkt?.harEpost
-
-            if (source.kontaktpunkt?.harTelefon != null)
-                destination.kontaktpunkt.harTelefon = source.kontaktpunkt?.harTelefon
-        }
-        if (source.omfang != null) {
-            if (source.omfang?.tekst != null)
-                destination.omfang.tekst = source.omfang?.tekst
-            if (source.omfang?.uri != null)
-                destination.omfang.uri = source.omfang?.uri
-        }
-        if (source.gyldigFom != null) {
-            destination.gyldigFom = source.gyldigFom
-        }
-
-        if (destination.endringslogelement == null) {
-            destination.endringslogelement = Endringslogelement()
-        }
-        if (source.endringslogelement != null) {
-            destination.endringslogelement.brukerId = source.endringslogelement.brukerId
-            destination.endringslogelement.endringstidspunkt = source.endringslogelement.endringstidspunkt
-        }
-        if (source.tillattTerm != null && source.tillattTerm.isNotEmpty()) {
-            destination.tillattTerm = source.tillattTerm
-        }
-        if (source.frarådetTerm != null && source.frarådetTerm.isNotEmpty()) {
-            destination.frarådetTerm = source.frarådetTerm
-        }
-
-        return destination
+        return ResponseEntity(HttpStatus.CONFLICT)
```

Comment:
den "else" er i veldig ulogisk sted. Eneste måte man kan havne her er hvis det er ønsket publisert, men validasjon feilet

Hva tenker du om dette:
```
if (publish && !isValid){
   return conflict
}
save
return ok
```

Reply:
valideringsfeil skulle være 400, Conflict 409 er litt annerledes - den sier brukeren et det er noe med nåværende server state. (for eksempel, du kunne si at du vil endre fra a til b, men server hadde allerede state c). Ja, vår Patch sjekker ikke server state, men uansett, vi validerer input, derfor error skulle være 400 og ikke 409

Reply:
Fixed